src/main.cpp

```diff
@@ -2313,4 +2320,124 @@ void vcRenderWindow(vcState *pProgramState)
 
   vcSettingsUI_Show(pProgramState);
   vcModals_DrawModals(pProgramState);
+
+#if UDPLATFORM_WINDOWS
+  if (pProgramState->settings.window.useNativeUI)
+  {
+    if (pProgramState->fileDialog.showDialog)
+    {
+      HRESULT hr;
+      IFileDialog *pFileOpen;
+
+      // Create the FileOpenDialog object.
+      if (pProgramState->fileDialog.loadOnly)
+        hr = CoCreateInstance(CLSID_FileOpenDialog, NULL, CLSCTX_ALL, IID_IFileOpenDialog, reinterpret_cast<void**>(&pFileOpen));
+      else
+        hr = CoCreateInstance(CLSID_FileSaveDialog, NULL, CLSCTX_ALL, IID_IFileSaveDialog, reinterpret_cast<void**>(&pFileOpen));
```

Comment:
HRESULT hr = CoCreateInstance(pProgramState->fileDialog.loadOnly ? CLSID_FileOpenDialog : CLSID_FileSaveDialog, NULL, CLSCTX_ALL, IID_IFileOpenDialog, reinterpret_cast<void**>(&pFileOpen));

Reply:
There are 2 variables that are changed here so rather than double ternary I split it out